File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string ans = "";
        int open = 0;
        for(char ch : s){
            if( ch == '(' && open++ > 0) 
                ans += ch;
            if(ch == ')' && open-- > 1)
                ans += ch; 
        }
        return ans;
    }
};

File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        StringBuilder ans = new StringBuilder();
        int open = 0;
        for(char ch : s.toCharArray()){
            if( ch == '(' && open++ > 0) 
                ans.append("(");
            if(ch == ')' && open-- > 1)
                ans.append(")"); 
        }
        return ans.toString();
    }
}